UnitsNet/CustomCode/UnitParser.cs

```diff
@@ -37,7 +37,8 @@ internal static class UnitParser
         internal static TUnit ParseUnit<TUnit>([NotNull] string str,
             [CanBeNull] IFormatProvider formatProvider,
             [NotNull] ParseUnit<TUnit> parseUnit,
-            [NotNull] Func<TUnit, TUnit, TUnit> add)
+            [NotNull] Func<TUnit, TUnit, TUnit> add,
+            Type unitType)
```

Comment:
How about adding another generic parameter instead? And renaming `TUnit` to `TQuantity`, as per recent discussion on naming conventions in #260.
Suggestion: `ParseUnit<TQuantity, TUnit>()`